src/main/scala/codecheck/github/models/Search.scala

```diff
@@ -0,0 +1,74 @@
+package codecheck.github.models
+
+import org.json4s.JValue
+import org.json4s.JArray
+
+sealed abstract class SearchSort(val name: String) {
+ override def toString = name
+}
+object SearchSort {
+  //for serachRepositories
+  case object stars extends SearchSort("stars")
+  case object forks extends SearchSort("forks")
+  case object updated extends SearchSort("updated")
+
+  //for searchCode
+  case object indexed extends SearchSort("indexed")
+
+  //for searchIssues
+  case object comments extends SearchSort("comments")
+  case object created extends SearchSort("created")
+  //case object updated extends SearchSort("updated")
+
+  //for searchUser
+  case object followers extends SearchSort("followers")
+  case object repositories extends SearchSort("repositories")
+  case object joined extends SearchSort("joined")
+}
+
+case class SearchInput (
+ q: String,
```

Comment:
The following is my implementation of an Input class very similar to this one.

```scala
case class IssueListOption4Repository(
  milestone: Option[MilestoneSearchOption] = None,
  state: IssueState = IssueState.open,
  assignee: Option[String] = None,
  creator: Option[String] = None,
  mentioned: Option[String] = None,
  labels: Seq[String] = Nil,
  sort: IssueSort = IssueSort.created,
  direction: SortDirection = SortDirection.desc,
  since: Option[DateTime] = None
) {
    def q = "?" + (if (!milestone.isEmpty) (milestone map (t => s"milestone=$t&")).get else "") +
      s"state=$state" +
      (if (!assignee.isEmpty) (assignee map (t => s"&assignee=$t")).get else "") +
      (if (!creator.isEmpty) (creator map (t => s"&creator=$t")).get else "") +
      (if (!mentioned.isEmpty) (mentioned map (t => s"&mentioned=$t")).get else "") +
      (if (!labels.isEmpty) "&labels=" + labels.mkString(",") else "") +
      s"&sort=$sort" +
      s"&direction=$direction" +
      (if (!since.isEmpty) (since map ("&since=" + _.toString("yyyy-MM-dd'T'HH:mm:ss'Z'"))).get else "")
 }
```

I think it may be beneficial to take in parameters and automatically generate the query string rather than having the user look up the format for it and writing it themselves.